src/api_server.py

```diff
@@ -47,15 +58,18 @@ class PredictResponse(BaseModel):
 
 @app.on_event("startup")
 def startup_event():
-    global classifier
+    global classifier_qa
+    global classifier_seq2seq
     global prediction_log_file
+
     logger.add(cnf.api_log_path, rotation="1 day", retention="7 days")
     logger.info("Starting API server")
 
-    classifier = BenefitsAndDrawbacksExtractor(
-        model_qa=cnf.model_qa,
-        batch_size=cnf.batch_size,
-        model_seq2seq=cnf.model_seq2seq
+    classifier_qa = QuestionAnsweringPredictor(
+        qa_model_name=cnf.model_name_qa, batch_size=cnf.batch_size, cuda_device_no=1
+    )
+    classifier_seq2seq = Seq2SeqPredictor(
+        model_name_seq2seq=cnf.model_name_seq2seq, cuda_device_no=1
```

Comment:
The variables `classifier_qa` and `classifier_seq2seq` can be deleted. Instead, simply set the appropriate keys in the `PREDICTORS` dictionary. Also `cuda_device_no` should be 0:

```python
    PREDICTORS["qa"] = QuestionAnsweringPredictor(
        qa_model_name=cnf.model_name_qa, batch_size=cnf.batch_size, cuda_device_no=0
    )
    PREDICTORS["seq2seq"] = Seq2SeqPredictor(
        model_name_seq2seq=cnf.model_name_seq2seq, cuda_device_no=0
    )
```

Reply:
I was wondering about that exactly, I was not sure when the classes will be instantiated. When we do a call or at event startup. Thanks for the info